kustomize/data-plane/README.md

```diff
@@ -0,0 +1,37 @@
+# Data Plane deployment
+
+The Data Plane deployment script creates a subscription for the shard-operator using a given [index image](environment-properties.env).
+In addition, it creates necessary ConfigMap and Secret resources according to given environment variables specified 
+in [environment-properties.env](environment-properties.env) and [secret-properties.env](secret-properties.env). 
+Inspired by https://mbuffa.github.io/tips/20210720-kustomize-environment-variables/.
+
+## Installation
+1. Set required operator parameters:
+- The env vars for shard-operator deployment are specified in [environment-properties.env](environment-properties.env), 
+for instance (replace with the expected values for your environment):
+  ```
+    EVENT_BRIDGE_FLEET_SHARD_INDEX_IMAGE
+    EVENT_BRIDGE_SSO_URL=http://keycloak:8180/auth/realms/event-bridge-fm
+    EVENT_BRIDGE_MANAGER_URL=http://event-bridge:8080
+  ```
+  Please note that not providing a value in the properties file (such as for `EVENT_BRIDGE_FLEET_SHARD_INDEX_IMAGE`) 
+  makes kustomize use a value from the environment variable with the same name.
+
+  **Important:** The values set in the properties file take precedence over environment variables.
+- The secrets for shard-operator deployment are specified in [secret-properties.env](secret-properties.env):
```

Comment:
@winklerm What's the reason we need another approach to secret handling? Does Sealed Secrets not work for you?

Reply:
I wanted to avoid setting the sealed secrets private key on the QE cluster (and other clusters); in addition, this approach allows you to inject any secrets you need in your Jenkins job which seemed more flexible and secure.

I agree it is yet another approach, thinking about it again..

Reply:
@winklerm I'd really advocate for keeping a single, consistent approach to secret management across everything. We're a very small team and having one way that everyone understands for secret management will help us all out.

We could for example install the Sealed Secrets operator as part of the prerequisistes and then you simply configure the key required for your environment as part of the CI pipeline? So pattern would be:

1. prerequisites installs SealedSecrets
2. your key for ^ is installed via the setup part of your CI pipeline, with the value being maintained in Jenkins (or whatever else is secure)
3. We store the secrets for your environment here, using the same consistent approach as we do for all our other environments.

Reply:
Agreed, this pattern could work, will take a look.

Reply:
Now I realized what I like about the properties files with the env. vars and secrets - I like that they specify the "interface" - you do not need to go through several kustomize resource files to find all the places you need to adjust something. For a person who is not familiar with the deployment scripts, it seems to be more straightforward (could be just my personal opinion).